src/item.cpp

```diff
@@ -3965,6 +3984,19 @@ bool item::ammo_consume( int qty ) {
     return false;
 }
 
+itype_id item::ammo_current() const
+{
+    if( is_ammo() ) {
+        return typeId();
```

Comment:
> That's why I came up with string_id<T>::NULL_ID

I did come across this but wasn't sure as to the usage. This can be used as a replacement for checking `id` == `null`?